ga-aio-projects/worker-cc-photoshop/src/dx-asset-compute-worker-1/actions/worker-cc-photoshop/index.js

```diff
@@ -0,0 +1,125 @@
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const { worker } = require('@adobe/asset-compute-sdk');
+const sdk = require('@adobe/aio-lib-photoshop-api');
+const libFiles = require('@adobe/aio-lib-files');
+const { v4: uuidv4 } = require("uuid");
+const { AioLibFilesMock } = require("../../lib/mock-aio-lib-files");
+const path = require("path");
+const { downloadFile } = require('@adobe/httptransfer');
+
+/**
+ * Acquire authorization
+ * 
+ * @param {*} params Worker parameters
+ * @returns {{orgId, clientId, accessToken}} authorization
+ */
+function getAuthorization(params) {
+    let orgId, clientId, accessToken;
+    if (process.env.WORKER_TEST_MODE) {
+        orgId = "test-ims-org-id";
+        clientId = "test-client-id";
+        accessToken = "test-access-token";
+    } else {
+        ({ orgId, clientId, accessToken } = params.auth);
+    }
+
+    if (!orgId || !clientId || !accessToken) {
+        throw Error("Request is missing authorization information");
+    }
+
+    return { orgId, clientId, accessToken };
+}
+
+/**
+ * Set up Photoshop Action Options
+ * 
+ * Resolve file types, add actionName parameter and 
+ * set up options for Photoshop Actions Api request
+ * 
+ * @param {*} client Photoshop Api Client instance
+ * @param {Object} instructions Rendition instructions
+ * @returns {Object} options for Photoshop Actions Api request
+ */
+async function setuphotoshopActionsOptions(client, instructions, files) {
+    if (!instructions || !instructions.photoshopActions) {
+        throw Error("Photoshop Action url not provided");
+    }
+
+    // workaround for action files from AEM
+    // we must download the action file and add the `.atn`
+    // extension to the file so Photoshop Service can
+    // recognize it
+    let photoshopActions = instructions.photoshopActions;
+    let ext;
+    try {
+        ext = path.extname(photoshopActions).substring(1).toLowerCase();
+    } catch (err) {
+    }
+    if (!ext) {
+        const tempActionFilename = `${uuidv4()}_temp.atn`;
+        const aioLibActionFilename = `${uuidv4()}/photoshopActions.atn`;
+        await downloadFile(photoshopActions, tempActionFilename);
+        await files.copy(tempActionFilename, aioLibActionFilename, { localSrc: true });
+        photoshopActions = aioLibActionFilename;
+        
+    }
+    const options = {
+        actions: [{
+            href: photoshopActions
+        }]
+    }
```

Comment:
Missin semicolon - either put them everywhere or nowhere
```suggestion
    };
```